src/infrastructure/apiRequest.ts

```diff
@@ -351,9 +351,10 @@ async function apiRequest(requestConfig: RequestConfig, cacheTTL?: number): Prom
       break;
     }
 
-    // Wait before retry (except on last attempt)
+    // Wait before retry (except on last attempt) with exponential backoff
     if (i < config.requestAttempts) {
-      await sleep(1500);
+      const delay = config.requestRetryDelays[i - 1] || 1500;
```

Comment:
This array doesn't scale if we increase TAGOIO_REQUEST_ATTEMPTS via env vars. It would be cleaner and safer to just calculate the delay math inside the loop.

We can replace the array lookup with this:
```javascript
// 1.5s, 3s, 6s, 12s... capped at 30s
const delay = Math.min(1500 * Math.pow(2, i - 1), 30000);
```